plugins/altstatusbar.koplugin/main.lua

```diff
@@ -0,0 +1,271 @@
+thanks to @poire-z
+That top status bar is not managed by KOReader, but by crengine (the engine thats renders EPUB files).
+You can tweak a bit of what it will show/not show by editing koreader/data/cr3.ini, these bits should apply to it:
+
+crengine.page.header.chapter.marks=1
+crengine.page.header.font.color=0xFF000000
+crengine.page.header.font.face=Noto Sans
+crengine.page.header.font.size=22
+
+window.status.battery=1
+window.status.battery.percent=0
+window.status.clock=0
+window.status.line=0
+window.status.pos.page.count=1
+window.status.pos.page.number=1
+window.status.pos.percent=0
+window.status.title=1
+]]
+
+--if true then return { disabled = true, } end
+
+local Device = require("device")
+local Event = require("ui/event")
+local ConfirmBox = require("ui/widget/confirmbox")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local logger = require("logger")
+local _ = require("gettext")
+
+local AltStatusBar = WidgetContainer:new{
+    name = "altstatusbar",
+    is_doc_only = true,
+}
+
+function AltStatusBar:isEnabled()
+    return G_reader_settings:readSetting("copt_status_line") == 0
+end
+
+function AltStatusBar:onReadSettings()
+    if self.document.provider == "crengine" then
+        self.enabled = self:isEnabled()
+        self.title = G_reader_settings:readSetting("cre_header_title") or 1
+        self.clock = G_reader_settings:readSetting("cre_header_clock") or 1
+        self.page_number = G_reader_settings:readSetting("cre_header_page_number") or 1
+        self.page_count = G_reader_settings:readSetting("cre_header_page_count") or 1
+        self.reading_percent = G_reader_settings:readSetting("cre_header_reading_percent") or 0
+        self.battery = G_reader_settings:readSetting("cre_header_battery") or 1
+        self.battery_percent = G_reader_settings:readSetting("cre_header_battery_percent") or 0
+        self.chapter_marks = G_reader_settings:readSetting("cre_header_chapter_marks") or 1
+
+        self.ui.document._document:setIntProperty("window.status.title", self.title)
+        self.ui.document._document:setIntProperty("window.status.clock", self.clock)
+        self.ui.document._document:setIntProperty("window.status.pos.page.number", self.page_number)
+        self.ui.document._document:setIntProperty("window.status.pos.page.count", self.page_count)
+        self.ui.document._document:setIntProperty("crengine.page.header.chapter.marks", self.chapter_marks)
+        self.ui.document._document:setIntProperty("window.status.battery", self.battery)
+        self.ui.document._document:setIntProperty("window.status.battery.percent", self.battery_percent)
+        self.ui.document._document:setIntProperty("window.status.pos.percent", self.reading_percent)
+
+        UIManager:broadcastEvent(Event:new("SetStatusLine", self.document.configurable.status_line, false))
```

Comment:
Is that needed? Isn't it already done by the readercoptlistener snippet I pasted above?

Reply:
Yes, can Line 65 can be deleted.